### PR TITLE
Fixes some altclick problems.

### PR DESCRIPTION
**Pull request by TechnoAlchemisto**
This should work, it's basically taken from what @oranges (optimumtact?) told me to do on my sabre quickdraw PR.

### CODE REVIEW

**Comment by optimumtact** [comment]
optimumtact on github
maybe I could be your gf tonight brother

**Comment by optimumtact** [comment]
This should be right up the top of the type heirarchy, under what circumstances would any alt click code expect non living mobs to be clicking on them?

File: code/game/objects/items/stacks/stack.dm [hunk 1]
```diff
@@ -234,7 +234,7 @@
 	return
 
 /obj/item/stack/AltClick(mob/living/user)
-	if(user.incapacitated())
+	if(!ishuman(user) || !user.canUseTopic(src))
```
Comment (TechnoAlchemisto) [inline code/game/objects/items/stacks/stack.dm]: good catcherino

Comment (optimumtact) [inline code/game/objects/items/stacks/stack.dm]: var/speciescheck = if(!ishuman(user) || !isdrone(user))
    var/usecheck = !user.canUseTopic(src)
    if(speciescheck || usecheck)
        ...failure case

File: code/game/objects/items/stacks/stack.dm [hunk 2]
```diff
@@ -234,7 +234,9 @@
 	return
 
 /obj/item/stack/AltClick(mob/living/user)
-	if(user.incapacitated())
+	var/speciescheck = !ishuman(user) || !isdrone(user)
```
Comment (ChangelingRain) [inline code/game/objects/items/stacks/stack.dm]: Prevents true devils and anything else that has hands from splitting stacks.

Comment (TechnoAlchemisto) [inline code/game/objects/items/stacks/stack.dm]: is there a check for things that have hands

Comment (X-TheDark) [inline code/game/objects/items/stacks/stack.dm]: All mobs have hands, at least defined, as per https://github.com/tgstation/tgstation/blob/master/code/modules/mob/mob_defines.dm#L82

You'd need to do something use procs `get_active_held_item()` (or `get_inactive_held_item()`, not sure which one precisely, @RemieRichards should be able to tell you, what with the mob hand rework and all that).

Comment (TechnoAlchemisto) [inline code/game/objects/items/stacks/stack.dm]: To be fair I don't think anything outside of humans and drones are going to have to utilize the alt click splitting feature anyways.  There are other ways of splitting stacks after all. If, for some reason, a true devil decides to start up a building project.

Comment (optimumtact) [inline code/game/objects/items/stacks/stack.dm]: just check is_living

Comment (TechnoAlchemisto) [inline code/game/objects/items/stacks/stack.dm]: where do I put that check?

Comment (optimumtact) [inline code/game/objects/items/stacks/stack.dm]: replace the species check with is_living

Comment (optimumtact) [inline code/game/objects/items/stacks/stack.dm]: no you want to invert the condition i.e !isliving(mob)

Comment (TechnoAlchemisto) [inline code/game/objects/items/stacks/stack.dm]: oh, duh, my bad.

File: code/game/objects/items/stacks/stack.dm [hunk 3]
```diff
@@ -234,7 +234,8 @@
 	return
 
 /obj/item/stack/AltClick(mob/living/user)
-	if(user.incapacitated())
+	var/usecheck = !user.canUseTopic(src)
+	if(!isliving(user) || usecheck)
```
Comment (phil235) [inline code/game/objects/items/stacks/stack.dm]: !istype(user)  since it's already typecasted in the argument.

**Pull request merged**